### ISSUE TITLE
app.config.file_watcher: undefined method `gsub' for #<Array:0x007f811792bec8> (NoMethodError)

### ISSUE THREAD

**Issue by xn** [activeadmin/activeadmin#3906]
I was installing `gem 'effective_assets'` when I ran into this. I opened a now closed issue over there:

https://github.com/code-and-effect/effective_assets/issues/5

The stack trace:

```
rails c
/Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/file_update_checker.rb:128:in `escape': undefined method `gsub' for #<Array:0x007f811792bec8> (NoMethodError)
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/file_update_checker.rb:122:in `block in compile_glob'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/file_update_checker.rb:121:in `each'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/file_update_checker.rb:121:in `map'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/file_update_checker.rb:121:in `compile_glob'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/file_update_checker.rb:40:in `initialize'
	from /Users/xn/.rvm/gems/ruby-2.2.1/bundler/gems/activeadmin-8968efcc51cc/lib/active_admin/application.rb:280:in `new'
	from /Users/xn/.rvm/gems/ruby-2.2.1/bundler/gems/activeadmin-8968efcc51cc/lib/active_admin/application.rb:280:in `block in attach_reloader'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/lazy_load_hooks.rb:36:in `call'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/lazy_load_hooks.rb:36:in `execute_hook'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/lazy_load_hooks.rb:45:in `block in run_load_hooks'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/lazy_load_hooks.rb:44:in `each'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/lazy_load_hooks.rb:44:in `run_load_hooks'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/application/finisher.rb:62:in `block in <module:Finisher>'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/initializable.rb:30:in `instance_exec'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/initializable.rb:30:in `run'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/initializable.rb:55:in `block in run_initializers'
	from /Users/xn/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:226:in `block in tsort_each'
	from /Users/xn/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:348:in `block (2 levels) in each_strongly_connected_component'
	from /Users/xn/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:429:in `each_strongly_connected_component_from'
	from /Users/xn/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:347:in `block in each_strongly_connected_component'
	from /Users/xn/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:345:in `each'
	from /Users/xn/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:345:in `call'
	from /Users/xn/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:345:in `each_strongly_connected_component'
	from /Users/xn/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:224:in `tsort_each'
	from /Users/xn/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/tsort.rb:203:in `tsort_each'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/initializable.rb:54:in `run_initializers'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/application.rb:352:in `initialize!'
	from /Users/xn/sites/contraption/education/config/environment.rb:5:in `<top (required)>'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `block in require'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:240:in `load_dependency'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/activesupport-4.2.1/lib/active_support/dependencies.rb:274:in `require'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/application.rb:328:in `require_environment!'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:142:in `require_application_and_environment!'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:67:in `console'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/xn/.rvm/gems/ruby-2.2.1/gems/railties-4.2.1/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:4:in `require'
	from bin/rails:4:in `<main>'
```

**Comment by xn** [comment]
The offending commit:

https://github.com/activeadmin/activeadmin/commit/b18d8cc8ce114af878dd997d7d2772ae44ed2b0d

(the parent commit does not generate this error.)
`gem 'activeadmin', github: 'activeadmin', ref: '3f916d6'`

**Comment by xn** [comment]
Effective Assets isn't really doing anything interesting with AA integration:
https://github.com/code-and-effect/effective_assets/search?utf8=%E2%9C%93&q=ActiveAdmin

**Comment by xn** [comment]
And as you can see from the trace, this is the offending line:
https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/application.rb#L280

**Comment by timoschilling** [comment]
It's `effective_assets` fault and should be closed by code-and-effect/effective_assets#7, read over there for details.

**Comment by matt-riemer** [comment]
Hey I'm the author of effective_assets.

Just wanted to jump in here and say thank you to @xn for discovering this issue and @timoschilling for resolving it.

Thanks.

ActiveAdmin is awesome, keep up the great work!
